src/envs/base_env.py

```diff
@@ -87,22 +90,62 @@ def action_space(self) -> Box:
         raise NotImplementedError("Override me!")
 
     @abc.abstractmethod
+    def render_state_plt(
+            self,
+            state: State,
+            task: Task,
+            action: Optional[Action] = None,
+            caption: Optional[str] = None) -> matplotlib.figure.Figure:
+        """Render a state and action into a Matplotlib figure.
+
+        Like simulate, this function is not meant to be part of the
+        "final system", where the environment is the real world. It is
+        just for convenience, e.g., in test coverage.
+
+        For environments which don't use Matplotlib for rendering, this
+        function should be overriden to simply crash.
+
+        NOTE: Users of this method must remember to call `plt.close()`,
```

Comment:
if you wanna be fancy, we could try something like https://stackoverflow.com/questions/63508841/using-with-statement-in-python-to-auto-close-a-pyplot-figure

Reply:
ehhh

Reply:
yeah lol nvm